File: src/036/p036.go
package main

import (
	"fmt"
	"strconv"
)

func palindrome(input string) bool {
	for i := 0; i < len(input)/2; i++ {
		if input[i] != input[len(input)-i-1] {
			return false
		}
	}
	return true
}

func numberPalindrome(input, base int) bool {
	input64 := int64(input)
	number := strconv.FormatInt(input64, base)
	return palindrome(number)
}

func doubleBasePalindrome(input int) bool {
	return numberPalindrome(input, 10) && numberPalindrome(input, 2)
}

func sumDoublePalindromeBelow(limit int) int {
	sum := 0
	for i := 1; i < limit; i++ {
		if doubleBasePalindrome(i) {
			sum += i
		}
	}
	return sum
}

func solve() int {
	return sumDoublePalindromeBelow(1000000)
}

func main() {
	result := solve()
	fmt.Println(result)
}

File: src/036/p036.py
#!/usr/bin/env python3
def palindrome(x):
    return str(x) == str(x)[::-1]


def number_palindrome(n, base):
    if base == 2:
        binary = bin(n)[2:]
        return palindrome(binary)

    if base == 10:
        return palindrome(n)

    return False


def double_base_palindrome(x):
    return number_palindrome(x, 10) and number_palindrome(x, 2)


def sum_double_palindrome_numbers_below(limit):
    return sum((x for x in range(1, limit) if double_base_palindrome(x)))


def solve():
    return sum_double_palindrome_numbers_below(1000000)


if __name__ == '__main__':
    result = solve()
    print(result)
